### PR TITLE
Follows and My Timelines for follows

### LINKED ISSUE #46: User timeline
Timeline containing all tweets from a user, whether they follow you and whether your follow them,

### LINKED ISSUE #47: My timeline
View twits from you and people you follow.

### LINKED ISSUE #66: Add follows to Entities.
We are currently missing follows which is needed for both the Api and the SSR. 
It can be implemented as a many to many relation.

### PR DESCRIPTION
**Pull request by Hjaltesorgenfrei**
Adds the Follow many to many relation, as well as timeline for the people you are following. 
fixes #47 
fixes #46
fixes #66

### CODE REVIEW

File: MiniTwit.Web.App/Views/Home/Index.cshtml [hunk 1]
```diff
@@ -3,54 +3,23 @@
...
-                    asp-controller="Home"
-                    asp-action="PostMessage">
-                    <p/><input asp-for="Text" size="60"/><!--
-                    --><input type="submit" value="share"/>
-                </form>
-            </div>
-        }
-    }
+    <div class="twitbox">
+        <h3>What's on your mind @* TODO: @ViewData["UserName"] *@?</h3>
```
Comment (jlndk) [inline MiniTwit.Web.App/Views/Home/Index.cshtml]: Should this todo be fixed, promoted to a issue or something else?

File: MiniTwit.Web.App/Views/Home/My_Timeline.cshtml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+@using Microsoft.AspNetCore.Identity
+@model Message
+@{
+    ViewData["Title"] = "Timeline";
+}
+
+<h2>@ViewData["Title"]</h2>
+<div>
+     <div class="twitbox">
+         <h3>What's on your mind @* TODO: @ViewData["UserName"] *@?</h3>
```
Comment (jlndk) [inline MiniTwit.Web.App/Views/Home/My_Timeline.cshtml]: Same about todo

**Review by jlndk — COMMENTED** [review]
Backend looks sweet, and frontend is okay.
Besides the comments i've added it is worth mentioning that there's quite a bit of code duplication between the views, which could perhaps be cleaned up.
Dunno if this should be resolved now or promoted to a issue

**Comment by Hjaltesorgenfrei** [comment]
I will fix the code duplication and fix the ToDo, it is quite simple and does not need a issue.

File: MiniTwit.Models.Test/MessageRepositoryTests.cs [hunk 3]
```diff
@@ -62,10 +62,10 @@ public async Task Read_messages_in_PubDate_order()
         {
             await Add_dummy_data(_userRepository, _messageRepository);
             var result = await _messageRepository.ReadAsync();
-            DateTime prev = DateTime.MinValue;
+            DateTime prev = DateTime.MaxValue;
             foreach (var message in result)
             {
-                Assert.True(DateTime.Compare(prev, message.PubDate) < 0);
+                Assert.Equal(1, DateTime.Compare(prev, message.PubDate));
```
Comment (Jakobis) [inline MiniTwit.Models.Test/MessageRepositoryTests.cs]: This should not use equal, but just true

File: MiniTwit.Models.Test/MessageRepositoryTests.cs [hunk 4]
```diff
@@ -77,10 +77,10 @@ public async Task ReadCount_messages_in_PubDate_order()
         {
             await Add_dummy_data(_userRepository, _messageRepository);
             var result = await _messageRepository.ReadCountAsync(12);
-            DateTime prev = DateTime.MinValue;
+            DateTime prev = DateTime.MaxValue;
             foreach (var message in result)
             {
-                Assert.True(DateTime.Compare(prev, message.PubDate) < 0);
+                Assert.Equal(1, DateTime.Compare(prev, message.PubDate));
```
Comment (Jakobis) [inline MiniTwit.Models.Test/MessageRepositoryTests.cs]: Same  as above

File: MiniTwit.Models/IMessageRepository.cs [hunk 5]
```diff
@@ -13,5 +13,6 @@ public interface IMessageRepository
         Task<Message> ReadAsync(int messageId);
         Task<Response> UpdateAsync(Message message);
         Task<Response> DeleteAsync(int messageId, bool force = false);
+        Task<IEnumerable<Message>> ReadAllMessagesFromFollowedAsync(int parse);
```
Comment (Jakobis) [inline MiniTwit.Models/IMessageRepository.cs]: Why is parse here?

File: MiniTwit.Models/MessageRepository.cs [hunk 6]
```diff
@@ -29,7 +29,7 @@ public async Task<IEnumerable<Message>> ReadAsync()
             var query = 
                 from m in _context.Messages
                 where m.Flagged <= 0
-                orderby m.PubDate
+                orderby m.PubDate descending
```
Comment (Jakobis) [inline MiniTwit.Models/MessageRepository.cs:32]: Sorry about that

**Review by Jakobis — CHANGES_REQUESTED** [review]

**Review by Duckapple — APPROVED** [review]
Seems good, things work, tests pass, I like

**Review by Jakobis — APPROVED** [review]
Simon told me to just approve without looking

**Pull request merged**